ggml/src/ggml-cuda/ggml-cuda.cu

```diff
@@ -2817,10 +2840,25 @@ static void evaluate_and_capture_cuda_graph(ggml_backend_cuda_context * cuda_ctx
                 }
 
                 static bool disable_fusion = (getenv("GGML_CUDA_DISABLE_FUSION") != nullptr);
-                if (!disable_fusion && ggml_cuda_can_fuse(cgraph, i, { GGML_OP_RMS_NORM, GGML_OP_MUL })) {
-                    ggml_cuda_op_rms_norm_fused(*cuda_ctx, node, cgraph->nodes[i+1]);
-                    i++;
-                    continue;
+                if (!disable_fusion) {
+                    if (ggml_cuda_can_fuse(cgraph, i, { GGML_OP_RMS_NORM, GGML_OP_MUL }, {})) {
+                        ggml_cuda_op_rms_norm_fused(*cuda_ctx, node, cgraph->nodes[i+1]);
+                        i++;
+                        continue;
+                    }
+
+                    if (ggml_cuda_can_fuse(cgraph, i, { GGML_OP_SCALE, GGML_OP_UNARY, GGML_OP_SCALE }, { GGML_UNARY_OP_TANH })) {
+                        ggml_tensor * src0 = node->src[0];
+                        float scale = ggml_get_op_params_f32(node, 0);
+
+                        i += 2; node = cgraph->nodes[i];
+
+                        ggml_set_op_params_f32(node, 1, scale);
+                        node->src[0] = src0;
+
+                        ggml_cuda_op_softcap(*cuda_ctx, node);
```

Comment:
It would be better to pass the values as parameters to the function instead of modifying the tensors. Or just pass both tensors to the function and let it extract the information it needs from them. Generally, tensors in a graph should be considered const.